### PR TITLE
Andy/validation

### PR DESCRIPTION
**Pull request by andyjianzhou**
- Waiting for #26 to be merged
- Add validation input
- Draft

- [x] solves #12

### CODE REVIEW

**Comment by andyjianzhou** [comment]
@Pradyuman7 
Made it work with data persistence, and organized code in a way that there are repeating lines of code for each POST request (same process)

- So I am thinking, because adding it to data.json is the same process, should I make a service/utility function to simply add it based off the API request name? Then, it will be very easy and we'll have a "service" layer to the codebase. Or is that an overkill in this case haha

**Comment by andyjianzhou** [comment]
Made it work with new and improved testcase

File: app.py [hunk 1]
```diff
@@ -24,25 +24,29 @@ def experience():
...
-
-    return jsonify({})
+        required_fields = ['title', 'company', 'start_date', 'end_date', 'description', 'logo']
+        
+        if not request.json:
+            return jsonify({'error': 'No data provided'}), 400
+        
+        missing_fields = [field for field in required_fields if field not in request.json]
+        if missing_fields:
+            return jsonify({'error': f'Missing required fields: {", ".join(missing_fields)}'}), 400
```
Comment (Pradyuman7) [inline app.py:37]: Generally, it is not a good idea to let a potential malicious user know what are the required fields and what are not. We could just say that some fields are missing, and not say what are those.
I don't think this necessary to do here, but just wanted to point out to ask what you think?

**Comment by Pradyuman7** [comment]
> @Pradyuman7 Made it work with data persistence, and organized code in a way that there are repeating lines of code for each POST request (same process)
> 
> * So I am thinking, because adding it to data.json is the same process, should I make a service/utility function to simply add it based off the API request name? Then, it will be very easy and we'll have a "service" layer to the codebase. Or is that an overkill in this case haha

I'd think an ovekill, but we could do it later as improvement

**Review by Pradyuman7 — APPROVED** [review]
LGTM 🥇

**Pull request merged**